crates/ruff_python_ast/src/source_code/locator.rs

```diff
@@ -0,0 +1,249 @@
+//! Struct used to efficiently slice source code at (row, column) Locations.
+
+use once_cell::unsync::OnceCell;
```

Comment:
(These files would be moved from `crates/ruff`, rather than copied. I think this roughly matches @MichaReiser's suggestion from the "split up crates" issue.)